ldelf/ta_elf.c

```diff
@@ -874,6 +923,53 @@ static void print_seg(size_t idx __maybe_unused, int elf_idx __maybe_unused,
 		 idx, width, va, width, pa, sz, flags_str, desc);
 }
 
+static bool get_next_in_order(struct ta_elf_queue *elf_queue,
```

Comment:
It's almost as you describe it. It's the lowest address which can accomodate the ELF + a random number of pages, the ELF is mapped at the end of the range and the preceding pages are free. This can lead to ELFs being mapped in different orders since a later ELF can fit in a previous range of free pages. I'll add a few words.

Reply:
```
Adds support in ldelf to load TAs using address space randomization if
CFG_TA_ASLR=y. Works as before with the kernel mode ELF loader with
two exceptions: 
- It falls back to disable ASLR for a particular ELF if it fails to map
  using ASLR.
- Each ELF is loaded with a randomized number of free pages in front.
  These free pages can be reused when loading other ELF permitting
  some ELF to become reordered in memory.

Reply:
Excellent description :) thanks.